app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -97,8 +97,23 @@ def on_shutdown
       @ticker.persist
     end
 
+    # We are reopening a private part of Job Iteration's API here, so we should
+    # ensure the method is still defined upstream. This way, in the case where
+    # the method changes upstream, we catch it at load time instead of at
+    # runtime while calling `super`.
+    unless private_method_defined?(:reenqueue_iteration_job)
+      raise 'JobIteration::Iteration#reenqueue_iteration_job must be defined'
```

Comment:
Maybe we could tell them to upgrade maintenance_tasks, because we should have a solution to this on our side.

Reply:
Yes, good idea 👍